rep-2004.rst

```diff
@@ -0,0 +1,521 @@
+REP: 2004
+Title: Package Quality Categories
+Author: William Woodall <william@openrobotics.org>
+Status: Active
+Type: Informational
+Content-Type: text/x-rst
+Created: 17-Dec-2019
+Post-History: 17-Dec-2019
+
+
+Abstract
+========
+
+This REP describes a set of categories which are meant to convey the quality, or at least the maturity, of packages in the ROS ecosystem.
+Inclusion in one category or another is based on the policies to which the packages adhere.
+The categories are meant to give some expectation as to the quality of a package and allows the maintainers to be more strict with some packages and less so with others.
+
+The purpose of these categories is not to enforce quality, but to set expectations for consumers of the packages and to encourage maintainers of the packages to document how their package's policies achieve that quality level.
```

Comment:
> The purpose of these categories is not to enforce quality, but to set expectations for consumers of the packages

That makes sense, although after an admittedly quick read I see no way for that quality level to be conveyed to consumers unless they visit the code repository in question. One of the ways we convey this type of thing to our users in Ubuntu is with [repository components](https://help.ubuntu.com/community/Repositories#Components); have you considered doing something similar?

Reply:
Yeah, basically I think we should have something like Ubuntu's repository components, or just a list of packages in each category, but I didn't want to a) mandate one in this REP or b) limit it to just one list (maybe other third-parties can have lists with different scope or strictness levels). At the end of the day, any list is editorial in nature, not 100% objective.